### PR TITLE
Loading fabric mods from different directories

### PR DESCRIPTION
**Pull request by Skidamek**
### Description
~~Replaces hardcoded `FMLPaths.MODSDIR` path with a configurable list of mods paths. This allows for greater flexibility, enabling other mods to seamlessly add other mods folders.~~

Adds `connector.additionalModLocations` property.
This allows to load extra fabric mods with

- the --connector.additionalModLocations program argument
- the -Dconnector.additionalModLocations jvm argument

Each takes a path separated by `,` e.g. `--connector.additionalModLocations=path/modA.jar,/some/path/modB.jar`.
It works as well on directories e.g. `--connector.additionalModLocations=some/path`.

### Motivation
This allows e.g. (future) [AutoModpack mod](https://modrinth.com/mod/automodpack) to load whole mods folder from completely different directory of modpack with Connector support. Tested to work fine with local dev version of the mod.

### CODE REVIEW

**Comment by Matyrobbrt** [comment]
I more meant making the additional mod dirs configurable via the sys prop :P

**Comment by Skidamek** [comment]
Now should work for directories as well as single jars

File: src/main/java/dev/su5ed/sinytra/connector/locator/ConnectorLocator.java [hunk 1]
```diff
@@ -176,6 +176,23 @@ private Stream<Path> scanClasspath() {
         }
     }
 
+    private Stream<Path> scanArguments() {
+        Stream<Path> pathStream = Arrays.stream(System.getProperty("connector.addMods", "").split(File.pathSeparator)).map(Path::of);
```
Comment (Matyrobbrt) [inline src/main/java/dev/su5ed/sinytra/connector/locator/ConnectorLocator.java:180]:
```suggestion
        Stream<Path> pathStream = Arrays.stream(System.getProperty("connector.additionalModLocations", "").split(",")).filter(s -> !s.isBlank()).map(Path::of);
```
Path separators are annoying and differ from system to system.
`additionalModLocations` is also a better name as `addMods` isn't very descriptive.

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

**Comment by Matyrobbrt** [comment]
Will be in Connector beta 34.

**Pull request merged**